pkg/services/alerting/eval_context.go

```diff
@@ -178,3 +179,47 @@ func getNewStateInternal(c *EvalContext) models.AlertStateType {
 
 	return models.AlertStateOK
 }
+
+// evaluateNotificationTemplateFields will treat the alert evaluation rule's name and message fields as
+// templates, and evaluate the templates using data from the alert evaluation's tags
+func (c *EvalContext) evaluateNotificationTemplateFields() error {
+	if len(c.EvalMatches) < 1 {
+		return nil
+	}
+
+	templateDataMap := buildTemplateDataMap(c.EvalMatches)
+	c.Rule.Message = evaluateTemplate(c.Rule.Message, templateDataMap)
+	c.Rule.Name = evaluateTemplate(c.Rule.Name, templateDataMap)
+
+	return nil
+}
+
+func evaluateTemplate(s string, m map[string]string) string {
+	for k, v := range m {
+		re := regexp.MustCompile(fmt.Sprintf(`\${%s}`, regexp.QuoteMeta(k)))
+		s = re.ReplaceAllString(s, v)
+	}
+
+	return s
+}
+
+// buildTemplateDataMap builds a map of alert evaluation tag names to a set of associated values (comma separated)
+func buildTemplateDataMap(evalMatches []*EvalMatch) map[string]string {
+	var result = map[string]string{}
+	for _, match := range evalMatches {
+		for tagName, tagValue := range match.Tags {
+			// skip duplicate values
+			rVal := regexp.MustCompile(fmt.Sprintf(`\b%s\b`, regexp.QuoteMeta(tagValue)))
```

Comment:
MustCompile panics on failure, so is only really meant when initializing a global var in something like `init()` of a package. Use `.Compile` instead and check the error.

Reply:
Good tip - thanks 😄  Just did a quick poke around and noticed the existing uses of this function in the codebase mostly ignores the err 😞  Something the linter should probably pick up..